handler/handler.go

```diff
@@ -87,18 +86,98 @@ func (h *Handler) DeleteAnnotation(w http.ResponseWriter, r *http.Request) {
 	}
 	uppList = uppList[:i]
 
+	h.writeAnnotationsRW(ctx, w, uppList, writeLog, oldHash, contentUUID)
+}
+
+// AddAnnotation adds an annotation for a specific content uuid.
+// It gets the annotations only from UPP skipping V2 annotations because they are not editorially curated.
+func (h *Handler) AddAnnotation(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "application/json")
+
+	contentUUID := vestigo.Param(r, "uuid")
+
+	tID := tidutils.GetTransactionIDFromRequest(r)
+	ctx := tidutils.TransactionAwareContext(context.Background(), tID)
+
+	writeLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", contentUUID)
+
+	if err := validateUUID(contentUUID); err != nil {
+		handleWriteErrors("Invalid content UUID", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	annotation := annotations.Annotation{}
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&annotation)
+	if err != nil {
+		handleWriteErrors("Error decoding request body", err, writeLog, w, http.StatusInternalServerError)
+	}
+
+	oldHash := r.Header.Get(annotations.PreviousDocumentHashHeader)
+
+	if err := validateUUID(annotation.ConceptId); err != nil {
+		handleWriteErrors("Invalid concept UUID", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	if err := validatePredicate(annotation.Predicate); err != nil {
+		handleWriteErrors(err.Error(), err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	uppList, err := h.getAnnotations(ctx, w, writeLog, contentUUID)
+	if err != nil {
+		handleErrors(err, writeLog, w)
+		return
+	}
+
+	conceptUUID := mapper.TransformConceptID("/" + annotation.ConceptId)
+	annotation.ConceptId = conceptUUID
+
+	uppList = h.canonicalizeAnnotations(uppList, writeLog)
+
+	for _, item := range uppList {
+		if conceptUUID == item.ConceptId && annotation.Predicate == item.Predicate {
+			writeLog.Debug("Annotation is already in list")
+			uppList = h.canonicalizeAnnotations(uppList, writeLog)
+			h.writeAnnotationsRW(ctx, w, uppList, writeLog, oldHash, contentUUID)
+			return
+		}
+	}
+
+	uppList = append(uppList, annotation)
+	uppList = h.canonicalizeAnnotations(uppList, writeLog)
+	h.writeAnnotationsRW(ctx, w, uppList, writeLog, oldHash, contentUUID)
+}
+
+func (h *Handler) getAnnotations(ctx context.Context, w http.ResponseWriter, writeLog *log.Entry, contentUUID string) ([]annotations.Annotation, error) {
```

Comment:
This new function does not shorten the written code. I suggest that it is removed.